### PR TITLE
New manage service

### PR DESCRIPTION
**Pull request by normanjaeckel**

### CODE REVIEW

File: pkg/setup/setup.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+}
+
+func writeContent(w io.Writer) error {
+	if _, err := w.Write([]byte("---\nfoo: bar\n")); err != nil {
+		return fmt.Errorf("writing content: %w", err)
+	}
+	return nil
+}
+
+func fileExists(p string) (bool, error) {
```
Comment (ostcar) [inline pkg/setup/setup.go:52]: There is a function `os.IsNotExist()` in the standard lib. It also contains an example how to check if an file exists: https://golang.org/pkg/os/#IsNotExist

**Review by ostcar — CHANGES_REQUESTED** [review]

File: pkg/setup/setup_test.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+func testDockerComposeYML(t *testing.T, d string) {
+	dcYml := path.Join(d, "docker-compose.yml")
+	if _, err := os.Stat(dcYml); errors.Is(err, os.ErrNotExist) {
+		t.Errorf("file %s does not exist, expected existance", dcYml)
+	}
+	dcYmlContent, err := os.ReadFile(dcYml)
+	if err != nil {
+		t.Errorf("reading file %s: %w", dcYml, err)
+	}
+	got := string(dcYmlContent[:])
```
[the remark this answers is not in the document]
Comment (normanjaeckel) [reply pkg/setup/setup_test.go]: Ok

File: pkg/setup/setup_test.go [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+	if got != defaultDockerComposeYml {
+		t.Errorf("checking content of YML file, expected `%s`, got `%s`", defaultDockerComposeYml, got)
```
[the remark this answers is not in the document]
Comment (normanjaeckel) [reply pkg/setup/setup_test.go]: ok

File: pkg/setup/setup.go [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+	"fmt"
+	"io"
+	"os"
+	"path"
+)
+
+const ymlFileName = "docker-compose.yml"
+
+// Setup creates YAML file for Docker Compose or Docker Swarm and secrets directory.
+func Setup(d string) error {
```
[the remark this answers is not in the document]
Comment (normanjaeckel) [reply pkg/setup/setup.go]: ok

File: pkg/setup/setup.go [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+	if err := createYMLFile(d); err != nil {
+		return fmt.Errorf("creating YML file at %s: %w", d, err)
+	}
+	return nil
+}
+
+func createYMLFile(d string) error {
+	p := path.Join(d, ymlFileName)
+
+	pExists, err := fileExists(p)
```
[the remark this answers is not in the document]
Comment (normanjaeckel) [reply pkg/setup/setup.go:50]: Ok.

File: pkg/setup/setup.go (hunk 1 above)
Comment (normanjaeckel) [reply pkg/setup/setup.go:71]: I read this but thex say:
` This function predates errors.Is. It only supports errors returned by the os package. New code should use errors.Is(err, os.ErrNotExist). `

**Comment by normanjaeckel** [comment]
Closed in favor of #57

**Pull request closed without merging**